### ISSUE TITLE
Should a support statement with flags get version_removed when unflagged?

### ISSUE THREAD

**Issue by gsnedders** [mdn/browser-compat-data#14777]
Seemingly we don't currently have consensus here. Sometimes they get `version_removed`, sometimes they don't. We should have a policy here, and enforce it with a lint (at least in the case when the `version_removed` would align with `version_added` without a flag).

For example, `api.AnimationTimeline` has:

```json
          "firefox": [
            {
              "version_added": "75"
            },
            {
              "version_added": "63",
              "flags": [
                {
                  "type": "preference",
                  "name": "dom.animations-api.timelines.enabled",
                  "value_to_set": "true"
                }
              ]
            }
          ],
```

Which results in Yari showing:

![MDN screenshot, including "From version 63: this feature is behind…"](https://user-images.githubusercontent.com/176218/152005305-781ca432-2edb-49df-b26f-3a9080ef4fa7.png)

Whereas `api.CanvasRenderingContext2D.createConicGradient` has:

```json
            "firefox": [
              {
                "version_added": "90"
              },
              {
                "version_added": "86",
                "version_removed": "90",
                "flags": [
                  {
                    "type": "preference",
                    "name": "canvas.createConicGradient.enabled"
                  }
                ]
              }
            ],
```

Which results in Yari showing:

![MDN screenshot, including "Removed in 90 and later"](https://user-images.githubusercontent.com/176218/152005337-82c3f98f-31d9-43b8-8df6-ec3d202af92e.png)

**Comment by queengooborg** [comment]
In BCD, we use a `version_removed` for the flags when the flag itself has been removed, rather than when it's enabled by default.  However, this isn't a defined policy, and I personally feel that it would be simpler just to mark the support behind a flag as removed once it's enabled by default.  We should definitely plan a policy and document it.

**Comment by gsnedders** [comment]
See also #16287 which added a guideline here.

That said, given the guideline says:

> Set `version_removed` to the earliest applicable version from these conditions. For example, if Chrome 90 enabled the flag by default, the feature could no longer be toggled off by disabling the flag in Chrome 96, and finally the flag was removed by Chrome 98, then set `version_removed` to `90` as it is the earliest applicable version.

How does this work with `preview` as a version? Do we treat `preview` as the earliest applicable version in all eternity, or do we change it following a release?

**Comment by queengooborg** [comment]
#16287 should probably have marked this issue closed, right?

In regards to `preview`, `preview` should always be considered the most recent (latest) version, since it's data for an upcoming/beta release.

**Comment by queengooborg** [comment]
I'm going to mark this as closed assuming that #16287 has covered this.
